cpp/scenario/gazebo/include/scenario/gazebo/Link.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This project is dual licensed under LGPL v2.1+ or Apache License.
+ *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SCENARIO_GAZEBO_LINK_H
+#define SCENARIO_GAZEBO_LINK_H
+
+#include <ignition/gazebo/Entity.hh>
+#include <ignition/gazebo/EntityComponentManager.hh>
+#include <ignition/gazebo/EventManager.hh>
+
+#include <array>
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace scenario {
+    namespace base {
+        struct Pose;
+        struct ContactData;
+    } // namespace base
+    namespace gazebo {
+        class Link;
+        class Model;
+    } // namespace gazebo
+} // namespace scenario
+
+class scenario::gazebo::Link
+{
+public:
+    Link();
+    virtual ~Link();
+
+    // ===========
+    // Gazebo Link
+    // ===========
+
+    uint64_t id() const;
+    bool initialize(const ignition::gazebo::Entity linkEntity,
+                    ignition::gazebo::EntityComponentManager* ecm,
+                    ignition::gazebo::EventManager* eventManager);
+    bool createECMResources();
+
+    // =========
+    // Link Core
+    // =========
+
+    std::string name() const;
+    std::array<double, 3> position() const;
+    std::array<double, 4> orientation() const;
+
+    std::array<double, 3> worldLinearVelocity() const;
+    std::array<double, 3> worldAngularVelocity() const;
+    std::array<double, 3> bodyLinearVelocity() const;
+    std::array<double, 3> bodyAngularVelocity() const;
+
+    std::array<double, 3> worldLinearAcceleration() const;
+    std::array<double, 3> worldAngularAcceleration() const;
+    std::array<double, 3> bodyLinearAcceleration() const;
+    std::array<double, 3> bodyAngularAcceleration() const;
```

Comment:
As in other comments, without the need to comment all the methods, I think that it make sense to try to do the effort to comment this methods that from our experience are the one more error-prone. cc @DanielePucci

Reply:
I totally agree. The main reason I am moving all the code to the `cpp` folder is to simplify getting files for Doxygen. It's in my TODO list and coming soon :) For the time being, do you think we can proceed forward without documentation. I want to make it ready when `refactoring` will be merged into `devel`.

Reply:
As long as the semantics of the methods is already clear to you, for me it is ok.